File: das_backend/src/main/java/ch/sbb/backend/admin/domain/service/ServicePointService.java
package ch.sbb.backend.admin.domain.service;

import ch.sbb.backend.admin.domain.model.ServicePoint;
import java.util.List;
import java.util.Optional;

public interface ServicePointService {

    List<ServicePoint> getAll();

    Optional<ServicePoint> findByUic(int uic);

    void updateAll(List<ServicePoint> servicePoints);

    void create(ServicePoint servicePoint);
}

File: das_backend/src/main/kotlin/ch/sbb/backend/servicepoints/domain/service/ServicePointService.kt
package ch.sbb.backend.servicepoints.domain.service

import ch.sbb.backend.servicepoints.domain.ServicePoint

interface ServicePointService {
    fun getAll(): List<ServicePoint>
    fun findByUic(uic: Int): ServicePoint?
    fun updateAll(servicePoints: List<ServicePoint>)
    fun create(servicePoint: ServicePoint)
}
